src/overlays/actors/ovl_Obj_Jgame_Light/z_obj_jgame_light.c

```diff
@@ -28,29 +33,173 @@ const ActorInit Obj_Jgame_Light_InitVars = {
+        } else {
+            this->unk_1AC = 0xC8;
+            this->unk_1B6 = 1;
+        }
+        if (this->unk_1A8 < 0.7f) {
+            this->unk_1A8 += 0.3f;
+        } else {
+            this->unk_1A8 = 1.0f;
+        }
+    } else if ((this->actor.colChkInfo.health & 8) != 0) {
+        if (this->unk_1AC >= 0x29) {
+            this->unk_1AC -= 0x28;
+        } else {
+            this->unk_1AC = -1;
+            if (this->unk_1A8 == 0.0f) {
+                this->unk_1B6 = 0;
+                this->actor.colChkInfo.health &= 0xFFFE;
+                this->actor.colChkInfo.health &= (s16)0xFFF7;
+            }
+        }
+        if (this->unk_1A8 > 0.3f) {
+            this->unk_1A8 -= 0.3f;
+        } else {
+            this->unk_1A8 = 0.0f;
+        }
+    }
+    if (this->unk_1A8 > 0.1f) {
+        func_800B9010(&this->actor, NA_SE_EV_TORCH - SFX_FLAG);
+    }
+    temp_a1 = (((s32)(Rand_ZeroOne() * 127.0f) + 0x80));
+    Lights_PointSetColorAndRadius(&this->lightInfo, temp_a1, (u32)(temp_a1 * 0.7f) & 0xFF, 0, this->unk_1AC);
+}
+
+void func_80C156C4(ObjJgameLight* this, GlobalContext* globalCtx) {
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
+
+void func_80C15718(ObjJgameLight* this, GlobalContext* globalCtx) {
+    u8 colChk;
+
+    if (((this->actor.colChkInfo.health & 1) != 0) && (((this->unk_1B8) & 1) == 0)) {
+        Audio_PlaySfxAtPos(&this->actor.projectedPos, NA_SE_EV_FLAME_IGNITION);
+        this->unk_1B8 = colChk = this->actor.colChkInfo.health;
+    }
+    if ((this->actor.colChkInfo.health & 2) != 0) {
+        this->actor.colChkInfo.health = this->actor.colChkInfo.health & 0xFFFD;
+        this->unk_1B2 = 0x12C & 0xFFFF;
+        this->unk_1B4 = 1;
+    } else if ((this->actor.colChkInfo.health & 4) != 0) {
+        this->actor.colChkInfo.health = this->actor.colChkInfo.health & 0xFFFB;
+        this->unk_1B2 = 0x12C;
+        this->unk_1B4 = 2;
+    }
+    if (this->unk_1B2 >= 0x10) {
+        this->unk_1B2 -= 0xF;
+    } else {
+        this->unk_1B2 = 0;
+    }
+}
+
+void ObjJgameLight_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjJgameLight* this = THIS;
+
+    func_80C15718(this, globalCtx);
+    func_80C15474(this, globalCtx);
+    func_80C156C4(this, globalCtx);
+    this->unk_1AE += 1;
+}
+
+void ObjJgameLight_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjJgameLight* this = THIS;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    func_8012C28C(globalCtx->state.gfxCtx);
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, &gObjectSyokudaiTypeSwitchCausesFlameDL);
+    if (this->unk_1B2 > 0) {
+        func_8012C2DC(globalCtx->state.gfxCtx);
+        if (this->unk_1B2 >= 0x100) {
```

Comment:
Ah, so this is an alpha, all associated literals should go in dec and this should be `> 255`